src/core/memory/memory.cc

```diff
@@ -15,13 +15,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#ifndef DISABLE_WARNINGS
 
+#ifdef USE_CUDA
```

Comment:
I mean the macro should be after include "singa/core/memory.h" or after include "singa/singa_config.h"